### PR TITLE
Stop testing PyPy 3.7, start testing 3.9

### PR DESCRIPTION
**Pull request by pquentin**
While revieweing https://github.com/python-trio/trio/pull/2488 from @harahu, I realized that the issue was in PyPy 3.7, not CPython. And given that PyPy has stopped supporting PyPy 3.7, I though a better fix could be to bump the PyPy versions we use. https://mail.python.org/archives/list/pypy-dev@python.org/thread/3W5HPNWKLYPVJSPFXJOSBHOIP3DCLSMY/

Let's try that!

### CODE REVIEW

**Comment by pquentin** [comment]
Never mind, it also happens on CPython:

```
FAILED ../../../../../../opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/trio/tests/test_ssl.py::test_SSLStream_generic[tls13-True] - trio.BrokenResourceError
FAILED ../../../../../../opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/trio/tests/test_ssl.py::test_ssl_bad_shutdown_but_its_ok[tls13] - trio.BrokenResourceError
FAILED ../../../../../../opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/trio/tests/test_ssl.py::test_ssl_https_compatibility_disagreement[tls13] - AssertionError: assert False
 +  where False = _is_eof(SSLError(1, '[SSL: KRB5_S_TKT_NYV] unexpected eof while reading (_ssl.c:2570)'))
 +    where SSLError(1, '[SSL: KRB5_S_TKT_NYV] unexpected eof while reading (_ssl.c:2570)') = BrokenResourceError().__cause__
 +      where BrokenResourceError() = <ExceptionInfo BrokenResourceError() tblen=3>.value
FAILED ../../../../../../opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/trio/tests/test_ssl.py::test_https_mode_eof_before_handshake[tls13] - trio.BrokenResourceError
FAILED ../../../../../../opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/trio/tests/test_ssl.py::test_SSLStream_generic[tls12-True] - trio.BrokenResourceError
FAILED ../../../../../../opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/trio/tests/test_ssl.py::test_ssl_bad_shutdown_but_its_ok[tls12] - trio.BrokenResourceError
FAILED ../../../../../../opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/trio/tests/test_ssl.py::test_ssl_https_compatibility_disagreement[tls12] - AssertionError: assert False
 +  where False = _is_eof(SSLError(1, '[SSL: KRB5_S_TKT_NYV] unexpected eof while reading (_ssl.c:2570)'))
 +    where SSLError(1, '[SSL: KRB5_S_TKT_NYV] unexpected eof while reading (_ssl.c:2570)') = BrokenResourceError().__cause__
 +      where BrokenResourceError() = <ExceptionInfo BrokenResourceError() tblen=3>.value
FAILED ../../../../../../opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/trio/tests/test_ssl.py::test_https_mode_eof_before_handshake[tls12] - trio.BrokenResourceError
```

**Comment by harahu** [comment]
@pquentin my understanding is that this PR should be put on ice until 3.7 EOL. Am I correct?

**Comment by harahu** [comment]
> there's no reason to test on a version that isnt actively developed or maintained, which pypy3.7 isn't.

I wasn't aware that pypy3.7 had reached EOL before Python 3.7. But it seems to check out. My bad.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -74,25 +74,24 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python: ['pypy-3.7', 'pypy-3.8', '3.7', '3.8', '3.9', '3.10', '3.11-dev']
+        python: ['pypy-3.8', '3.7', '3.8', '3.9', '3.10', '3.11-dev']
```
Comment (harahu) [inline .github/workflows/ci.yml:77]: Shouldn't `'pypy-3.9'` be added to the the list here?

**Comment by pquentin** [comment]
Sorry for leaving this unattended. The situation has changed a bit since I opened this. What I think we should do regarding PyPy is:

 * moving to the setup-python GitHub Actions that now supports PyPy (and is handled by the PyPy team)
 * start testing PyPy 3.9 and its nightly
 * decide what we do for PyPy 3.7. I think it makes sense to drop it as it's no longer supported by the PyPy team

Closing this as I would rather someone open a new pull request for the above. Thanks.

**Pull request closed without merging**